### ISSUE TITLE
Migrate barriers to product-selector

### ISSUE THREAD

**Issue by commuterjoy** [Financial-Times/next-product-selector#28]
## Phase 1

- [x] Add suspended barrier
- [x] Take out first-click-free from next-article (?) - @ifyio did this
- [x] Fix vary headers in product selector
- [x] Track which barriers are being served in graphite
- [x] Release Fastly change - one at a time ...
  - [x] Suspended, http://git.svc.ft.com/projects/NEXT/repos/fastly-config/pull-requests/112/diff
  - [x] Premium, Subscription etc.
  - [x] Trial
- [x] Make product selector respond to preflight barriers - Eg, content-classification etc.
- [x] Write some unit tests for product selector w/ sample responses
- [x] Add article barrier headline etc. aesthetics - _we decided not to go with an overlay aesthetic._
- [x] Add registration barrier #38 
- [x] QA - barrier types, first click free (privileged referrer)

## Phase 2

- [x] Verify country-code is being varied on in CDN
- [x] Remove spoor page view tracking from product selector
- [x] Take out all the barrier logic inside article - https://github.com/Financial-Times/next-article/pull/1181
    - [x] Deprecate next-barriers 
- [x] Point registered people at subscription barrier (i.e. upgrade their accounts)
- [x] QA - Check all the seg-ids etc. & check they exist in the report
- [x] Migrate corporate barriers to product selector (perhaps just corp overlay initially?)

**Comment by ifyio** [comment]
Yes I removed first click free messaging from beneath the article. I was told by @SarahHornsey that it was not supposed to be there. The messaging has been removed for a while now (since the days of Claire with her being aware of the removal). The only thing I removed yesterday was the remnants of the messaging like its setup in the controller. Note though @commuterjoy that I didn't remove the first click free functionality (i.e., the ability to see the first clicked article for free). I just removed the messaging on the article page that was behind by the `firstClickFree` feature flag.
